### DISCUSSION TITLE
Type annotations for Python interface

### DISCUSSION THREAD

**Discussion by tgbosch** [casadi/casadi#3631] [Q&A]
Hi,

I was wondering if there is an easy way to get at least partial type annotations to CasADi's Python interface?

I tried to autogenerate a stub file using Mypy's `stubgen`, but this did not turn out to provide meaningful type annotations out of the box.
If anybody has some ideas how to or maybe already has created partial type annotations for CasADi, I would be great to know!

Thanks in advance and thanks for CasADi in general!

**Comment by jbentvelsen** [comment]
I think this would be a great addition!

**Comment by jgillis** [comment]
This is probably more of a feature request for swig: https://github.com/swig/swig/issues/735
I feel like it's unmaintainable to do it manually
